File: P2-Linguagem_de_Programação_II/Roteiros de Aula - Java/Roteiro de Aula 08/Produto.java
package Roteiro de Aula 08;

public class Produto {
    private String nome;
    private int estoqueReal;
    private int estoqueIdeal;

    public Produto(String nome, int estoqueReal, int estoqueIdeal) {
        this.nome = nome;
        this.estoqueReal = estoqueReal;
        this.estoqueIdeal = estoqueIdeal;
    }

    public boolean verificarNecessidadeCompra() {
        double limiteMinimo = this.estoqueIdeal * 0.20;
        return this.estoqueReal < limiteMinimo;
    }

    public void registrarSaida(int quantidade) {
        if (quantidade <= this.estoqueReal) {
            this.estoqueReal -= quantidade;
            System.out.println("Venda realizada! Novo saldo de " + this.nome + ": " + this.estoqueReal);
        } else {
            System.out.println("ERRO: Operação Bloqueada para " + this.nome + "!");
            System.out.println("Risco de Estoque Negativo. Saldo disponível: " + this.estoqueReal);
        }
    }
}

public class SistemaEstoque {
    public static void main(String[] args) {
        Produto poteChocolate = new Produto("Pote Chocolate 2L", 15, 100);
        Produto picoleLimao = new Produto("Picolé de Limão", 50, 100);

        System.out.println("Chocolate precisa repor? " + poteChocolate.verificarNecessidadeCompra());
        System.out.println("Limão precisa repor? " + picoleLimao.verificarNecessidadeCompra());

        poteChocolate.registrarSaida(20);
    }
}

File: P2-Linguagem_de_Programação_II/Roteiros de Aula - Python/Roteiro de Aula 08/produto.py
class Produto:
    def __init__(self, nome, estoque_real, estoque_ideal):
        self._nome = nome
        self._estoque_real = estoque_real
        self._estoque_ideal = estoque_ideal

    def verificar_necessidade_compra(self):
        limite_minimo = self._estoque_ideal * 0.20
        return self._estoque_real < limite_minimo

    def registrar_saida(self, quantidade):
        if quantidade <= self._estoque_real:
            self._estoque_real -= quantidade
            print(f"Venda realizada! Novo saldo de {self._nome}: {self._estoque_real}")
        else:
            print(f"ERRO: Operação Bloqueada para {self._nome}!")
            print(f"Risco de Estoque Negativo. Saldo atual: {self._estoque_real}")

pote_chocolate = Produto("Pote Chocolate 2L", 15, 100)
picole_limao = Produto("Picolé de Limão", 50, 100)

print(f"Chocolate precisa comprar? {pote_chocolate.verificar_necessidade_compra()}") 
print(f"Limão precisa comprar? {picole_limao.verificar_necessidade_compra()}")  

pote_chocolate.registrar_saida(20) 